src/SchemaBuilder/SchemaBuilderDecorator.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ScrumWorks\OpenApiSchema\SchemaBuilder;
+
+use ReflectionClass;
+use ReflectionProperty;
+use ScrumWorks\OpenApiSchema\Exception\LogicException;
+use ScrumWorks\OpenApiSchema\ValueSchema\Builder\AbstractSchemaBuilder;
+use ScrumWorks\OpenApiSchema\ValueSchema\Builder\ArraySchemaBuilder;
+use ScrumWorks\OpenApiSchema\ValueSchema\Builder\BooleanSchemaBuilder;
+use ScrumWorks\OpenApiSchema\ValueSchema\Builder\EnumSchemaBuilder;
+use ScrumWorks\OpenApiSchema\ValueSchema\Builder\FloatSchemaBuilder;
+use ScrumWorks\OpenApiSchema\ValueSchema\Builder\HashmapSchemaBuilder;
+use ScrumWorks\OpenApiSchema\ValueSchema\Builder\IntegerSchemaBuilder;
+use ScrumWorks\OpenApiSchema\ValueSchema\Builder\MixedSchemaBuilder;
+use ScrumWorks\OpenApiSchema\ValueSchema\Builder\ObjectSchemaBuilder;
+use ScrumWorks\OpenApiSchema\ValueSchema\Builder\StringSchemaBuilder;
+
+class SchemaBuilderDecorator
+{
+    /**
+     * @var PropertySchemaDecoratorInterface[]
+     */
+    private array $propertyDecorators;
+
+    /**
+     * @var ClassSchemaDecoratorInterface[]
+     */
+    private array $classDecorators;
+
+    /**
+     * @param PropertySchemaDecoratorInterface[] $propertyDecorators
+     * @param ClassSchemaDecoratorInterface[] $classDecorators
+     */
+    public function __construct(array $propertyDecorators, array $classDecorators)
```

Comment:
tady se to vazne k myslence se kterou jsem si hral uz pri prvnim rebuildu, totiz jestli misto poli nejit spise cestou "middlewares" ktere by si to mezi sebou posilali

Reply:
V tomto pripade mi davaji vetsi smysl dekoratory, protoze se nejedna o transformaci vstupu na vystup, ale o modifikaci builderu dle ruznych kriterii.